robot_sdk/wrappers/ec_wrapper.py

```
@@ -1,101 +1,397 @@
+        """
 
-    def __init__(self,ip) -> None:
-        print("Successfully loaded the SDK for Elite EC series")
-        self.client = Owl_Ec_client(ip=ip)
-        print("Connected to the Robot Controller")
+        try:
+            print("Successfully loaded the SDK for Elite EC series")
+            self.client = Owl_Ec_client(ip=ip)
+            print(f"Connected to the Robot Controller at {ip}")
+        except Exception as e:
+            raise ConnectionError(f"Failed to connect to robot at {ip}: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)